clang/lib/Basic/Targets/PPC.h

```diff
@@ -359,6 +359,13 @@ class LLVM_LIBRARY_VISIBILITY PPCTargetInfo : public TargetInfo {
   bool isSPRegName(StringRef RegName) const override {
     return RegName.equals("r1") || RegName.equals("x1");
   }
+
+  // We support __builtin_cpu_supports/__builtin_cpu_is on targets that
```

Comment:
Musl targets are supposed to use a "musl" triple.  We already have Triple::isMusl(). If it's somehow possible for both isOSGlibc() and isMusl() to be true, I'd consider that a bug in isOSGlibc().

Reply:
Linux musl also supports `AT_HWCAP` `AT_HWCAP2`.

Reply:
We can certainly expand this in a follow-up patch, but since I don't really have a way to do any functional testing on a musl machine, I'll leave it out of this patch.